[torchx] Do not terminate parent process if exit code from child isn't valid

atuljangra opened the pull request:
Summary:
There's no reason to terminate the parent process trying to find the name of the signal received by the child process.
Let's make sure this is handled properly, which then will ensure that parent process can process child failures.

Test Plan: Unit tests.

Reviewed By: aaronenyeshi

Differential Revision: D50615419

aaronenyeshi commented:
/easycla

atuljangra commented:
Added the correct email address as well.

aaronenyeshi commented:
/easycla

aaronenyeshi commented:
> Added the correct email address as well.

That check is passing now, once all the signals pass, we can use pytorchbot to merge this PR.

atuljangra committed:
[torchx] Do not terminate parent process if exit code from child isn't valid (#111961)

Summary:

There's no reason to terminate the parent process trying to find the name of the signal received by the child process.
Let's make sure this is handled properly, which then will ensure that parent process can process child failures.

Test Plan: Unit tests.

Reviewed By: aaronenyeshi

Differential Revision: D50615419
